lib/bgm.dart

```diff
@@ -13,11 +13,11 @@ import 'package:flutter/widgets.dart';
 /// to keep running.
 class Bgm extends WidgetsBindingObserver {
   bool _isRegistered = false;
-  AudioCache audioCache;
-  AudioPlayer audioPlayer;
+  late AudioCache audioCache;
+  AudioPlayer? audioPlayer;
   bool isPlaying = false;
 
-  Bgm({AudioCache audioCache}) {
+  Bgm({AudioCache? audioCache}) {
```

Comment:
This could be refactored so that the `audioCache` can become final:

```suggestion
  Bgm({AudioCache? audioCache}) : this.audioCache = audioCache ?? AudioCache()
```